src/aviary/env_client.py

```diff
@@ -23,36 +30,132 @@ def __init__(
         self._request_headers = request_headers
         self._request_timeout = request_timeout
 
-    async def reset(self) -> tuple[list[Message], list[Tool]]:
+    async def _post(self, url: str, json: dict[str, Any]) -> httpx.Response:
         async with httpx.AsyncClient() as client:
             response = await client.post(
-                self._reset_request_url,
-                json=self._make_post_json(self.state),
+                url,
+                json=json,
                 params=self._request_params,
                 headers=self._request_headers,
                 timeout=self._request_timeout,
             )
             response.raise_for_status()
-            msgs, tools = response.json()
-            return MessagesAdapter.validate_python(msgs), ToolsAdapter.validate_python(
-                tools
-            )
+            return response
+
+    async def reset(self) -> tuple[list[Message], list[Tool]]:
+        response = await self._post(
+            self._reset_request_url, json=self._make_post_json(self.state)
+        )
+        msgs, tools = response.json()
+        return (
+            MessagesAdapter.validate_python(msgs),
+            ToolsAdapter.validate_python(tools),
+        )
 
     async def step(
         self, action: ToolRequestMessage
     ) -> tuple[list[Message], float, bool, bool]:
-        async with httpx.AsyncClient() as client:
-            response = await client.post(
-                self._step_request_url,
-                json=self._make_post_json(self.state) | {"action": action.model_dump()},
-                params=self._request_params,
-                headers=self._request_headers,
-                timeout=self._request_timeout,
-            )
-            response.raise_for_status()
-            messages, reward, done, truncated = response.json()
-            return MessagesAdapter.validate_python(messages), reward, done, truncated
+        response = await self._post(
+            self._step_request_url,
+            json=self._make_post_json(self.state) | {"action": action.model_dump()},
+        )
+        messages, reward, done, truncated = response.json()
+        return MessagesAdapter.validate_python(messages), reward, done, truncated
 
     @abstractmethod
     def _make_post_json(self, state: TEnvState) -> dict[str, Any]:
         """Extract values from state to sent as JSON for all reset/step POSTs."""
+
+
+class TaskEnvClientState(BaseModel):
+    env_id: str
```

Comment:
Can you add a `description` saying where this comes from? Something like "the ID of a started environment" or something